analytics/filesystem/file_module.go

```diff
@@ -103,13 +102,25 @@ func NewFileLogger(filename string) (analytics.PBSAnalyticsModule, error) {
 }
 
 func jsonifyAuctionObject(ao *analytics.AuctionObject) string {
-	type alias analytics.AuctionObject
+	var logEntry *logAuction
+	if ao != nil {
+		logEntry = &logAuction{
+			Status:               ao.Status,
+			Errors:               ao.Errors,
+			Request:              ao.Request,
+			Response:             ao.Response,
+			Account:              ao.Account,
+			StartTime:            ao.StartTime,
+			HookExecutionOutcome: ao.HookExecutionOutcome,
+		}
+	}
+
 	b, err := json.Marshal(&struct {
 		Type RequestType `json:"type"`
-		*alias
+		*logAuction
 	}{
-		Type:  AUCTION,
-		alias: (*alias)(ao),
+		Type:       AUCTION,
```

Comment:
Do you think `type` can be moved to inside of  `logAuction`? And similar for other structures. 
Is `logAuction struct` will be used for other purposes?

Reply:
This is a good idea. I hadn't tried. I coded the suggestion but found functional differences in the case where the log object is nil. With this current approach, a nil object would produce:

```
{
    "type": "/openrtb2/auction"
}
```

With moving `type` instead of `logAuction` I can no longer produce this result. A nil log object would still serialize all fields in their empty state:

```
{
    "type": "/openrtb2/auction",
    "Status": 0,
    "Errors": null,
    "Request": null,
    "Response": null,
    "Account": null,
    "StartTime": "0001-01-01T00:00:00Z",
    "HookExecutionOutcome": null
}
```

Adding the "omitempty" directive to each field would also not produce the same result.

Reply:
Oh I see. Just curious: do we need to return type only in case in case of nil log object?

Reply:
Yes. That's the current behavior. I'm hesitant to change it because I don't know know how folks are relying on it's current behavior. It's an interesting side effect of using json serialization with an embedded / wrapped type.